desc/basis.py

```diff
@@ -1800,17 +1800,17 @@ def _jacobi_body_fun(kk, d_p_a_b_x):
     # other edge cases
     out = jnp.where(n == 0, 1.0, out)
     out = jnp.where(n == 1, 0.5 * (2 * (alpha + 1) + (alpha + beta + 2) * (x - 1)), out)
-    return c * out
+    return (c * out).astype(jnp.float64)
 
 
 @_jacobi.defjvp
-def _jacobi_jvp(x, xdot):
-    (n, alpha, beta, x, dx) = x
-    (ndot, alphadot, betadot, xdot, dxdot) = xdot
+def _jacobi_jvp(dx, x, xdot):
+    (n, alpha, beta, x) = x
+    (*_, xdot) = xdot
     f = _jacobi(n, alpha, beta, x, dx)
     df = _jacobi(n, alpha, beta, x, dx + 1)
     # in theory n, alpha, beta, dx aren't differentiable (they're integers)
     # but marking them as non-diff argnums seems to cause escaped tracer values.
     # probably a more elegant fix, but just setting those derivatives to zero seems
     # to work fine.
-    return f, df * xdot + 0 * ndot + 0 * alphadot + 0 * betadot + 0 * dxdot
```

Comment:
I deleted redundant 0 multiplications because in some cases, this gives an error saying `float0 cannot be used in math operations...`

Reply:
I thought that was an issue before? but I guess if you ran it with the jax versions and this passes then this is fine

Reply:
Ah no, the problem there was different. Back then we thought the problem was for `zernike_radial`, but it was `root` and `root_scalar`